sdk/storage/Azure.Storage.Common/src/Shared/CompatSwitches.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.Storage
+{
+    internal static class CompatSwitches
+    {
+        private static bool? _disableRequestConditionsValidation;
+
+        public static bool DisableRequestConditionsValidation => _disableRequestConditionsValidation
```

Comment:
Does making this static potentially cause problems if multiple libraries access this value from different environments?

Reply:
No, this is an internal shared file, so each library will have its own copy of the `CompatSwitches` type